staging/src/k8s.io/apimachinery/pkg/api/validate/schema.go

```diff
@@ -36,12 +38,9 @@ func MaxLength(_ operation.Context, fldPath *field.Path, value, _ *string, max i
 
 // Required verifies that the specified value is not the zero-value for its
 // type.
-func Required[T comparable](_ operation.Context, fldPath *field.Path, value, _ *T) field.ErrorList {
-	if value != nil {
-		var zero T
-		if *value != zero {
-			return nil
-		}
+func Required[T any](_ operation.Context, fldPath *field.Path, value, _ T) field.ErrorList {
+	if !reflect.Indirect(reflect.ValueOf(value)).IsZero() {
```

Comment:
IIUC this can panic with a nil pointer being passed as `value` due to unsafe reflection operations. Go playground example: https://go.dev/play/p/otoITcc41Ps

Assuming nil pointers can be passed to here (I think they could be IIUC), this code might need validation before chaining reflect.Indirect() and IsZero() to handle these cases safely.